morpheus/stages/input/file_source.py

```diff
@@ -21,140 +21,126 @@
-        (Note: Applicable when all paths in `files` are local.)
-    recursive : bool, default = True
-        If true, events will be emitted for the files in subdirectories matching `input_glob`.
-        (Note: Applicable when all paths in `files` are local.)
-    queue_max_size : int, default = 128
-        Maximum queue size to hold the file paths to be processed that match `input_glob`.
-        (Note: Applicable when all paths in `files` are local.)
-    batch_timeout : float, default = 5.0
-        Timeout to retrieve batch messages from the queue.
-        (Note: Applicable when all paths in `files` are local.)
-    file_type : `morpheus.common.FileTypes`, optional, case_sensitive = False
+    sort : bool, default = False
+        If true, the list of files will be processed in sorted order.
+    file_type : morpheus.common.FileTypes, optional, case_sensitive = False
         Indicates what type of file to read. Specifying 'auto' will determine the file type from the extension.
         Supported extensions: 'csv', 'json', 'jsonlines' and 'parquet'.
-    repeat : int, default = 1, min = 1
-        Repeats the input dataset multiple times. Useful to extend small datasets for debugging.
-    filter_null : bool, default = True
-        Whether or not to filter rows with a null 'data' column. Null values in the 'data' column can cause issues down
-        the line with processing. Setting this to True is recommended.
-    parser_kwargs : dict, default = {}
+    parser_kwargs : dict, default = None
         Extra options to pass to the file parser.
+    max_files: int, default = -1
+        Max number of files to read. Useful for debugging to limit startup time. Default value of -1 is unlimited.
     """
 
     def __init__(self,
                  config: Config,
                  files: typing.List[str],
                  watch: bool = False,
                  watch_interval: float = 1.0,
-                 sort_glob: bool = False,
-                 recursive: bool = True,
-                 queue_max_size: int = 128,
-                 batch_timeout: float = 5.0,
+                 sort: bool = False,
                  file_type: FileTypes = FileTypes.Auto,
-                 repeat: int = 1,
-                 filter_null: bool = True,
-                 parser_kwargs: dict = None):
+                 parser_kwargs: dict = None,
+                 max_files: int = -1):
 
         super().__init__(config)
 
-        if not files:
+        if not files or len(files) == 0:
             raise ValueError("The 'files' cannot be empty.")
 
         if watch and len(files) != 1:
             raise ValueError("When 'watch' is True, the 'files' should contain exactly one file path.")
 
         self._files = list(files)
+        self._protocols = self._extract_unique_protocols()
+
+        if len(self._protocols) > 1:
+            raise ValueError("Accepts same protocol input files, but it received multiple protocols.")
+
```

Comment:
Raising an error if `self._files` is `None` or `[]`. We will get at least one value in the `self._protocols`, so i didn't put an extra check.

Reply:
The `max_files` flag takes effect only when set to a value greater than zero; otherwise, it is treated as continuous polling without any imposed limit. Default value is `-1`, so I thought raising an error or warn would not need. Let me know if you still want to add the warning message.